src/helpers/DataRequirementHelpers.ts

```diff
@@ -69,6 +77,28 @@ export function generateDetailedDateFilter(filter: DuringFilter): R4.IDataRequir
   };
 }
 
+/**
+ * Map a filter into a FHIR DataRequirement valueFilter extension
+ *
+ * @param filter the filter to map
+ * @returns extension for the valueFilter list of dataRequirement
+ */
+export function generateDetailedValueFilter(filter: Filter): R4.IExtension {
+  if (filter.type === 'notnull') {
+    const notnullFilter = filter as NotNullFilter;
+    return {
+      url: 'http://example.com/dr-value',
+      extension: [
+        { url: 'dr-value-attribute', valueString: notnullFilter.attribute },
+        { url: 'dr-value-filter', valueString: 'not null' }
+      ]
+    };
+  } else {
+    console.warn(`Detailed value filter is not yet supported for filter type ${filter.type}`);
+    return {};
```

Comment:
🤔 Would it be better to return `null` and check the output of this function to make sure a result came back before putting it in the report? This would make sure we don't end up putting an empty object where there should be a complete extension.